### PR TITLE
Unify logging by using logrus for jetstream logs

### PR DESCRIPTION
**Pull request by dvob**
I guess tests for the logging is rather unusual so I omitted tests for this change.

* [x] I have added Go unit tests or [Complement integration tests](https://github.com/matrix-org/complement) for this PR _or_ I have justified why this PR doesn't need tests
* [x] Pull request includes a [sign off below using a legally identifiable name](https://matrix-org.github.io/dendrite/development/contributing#sign-off) _or_ I have already signed off privately

Signed-off-by: `David Schneider <dsbrng25b@gmail.com>`

### CODE REVIEW

File: setup/jetstream/nats.go [hunk 1]
```diff
@@ -53,7 +53,7 @@ func (s *NATSInstance) Prepare(process *process.ProcessContext, cfg *config.JetS
 		if err != nil {
 			panic(err)
 		}
-		s.ConfigureLogger()
+		s.SetLogger(NewLogAdapter(), false, false)
```
Comment (S7evinK) [inline setup/jetstream/nats.go:56]: Maybe worth to explain that `false, false` is `debugFlag` and `traceFlag`

File: setup/jetstream/log.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+		entry: logrus.StandardLogger().WithField("component", "jetstream"),
+	}
+}
+
+func (l *LogAdapter) Noticef(format string, v ...interface{}) {
+	l.entry.Infof(format, v...)
+}
+
+func (l *LogAdapter) Warnf(format string, v ...interface{}) {
+	l.entry.Errorf(format, v...)
```
Comment (S7evinK) [inline setup/jetstream/log.go:25]: Looks like there are a few copy and paste mistakes? e.g. `Warnf` logging on `Errorf` and `Fatalf` only on `Errorf`?

**Review by S7evinK — CHANGES_REQUESTED** [review]

File: setup/jetstream/nats.go (hunk 1 above)
Comment (dvob) [reply setup/jetstream/nats.go]: Changed this to use the values from the NATS options which mimics the exact behavior of ConfigureLogger which was used before.

**Comment by dvob** [comment]
@S7evinK Thank you for the review. 

Yes these there were some copy/paste mistakes which I fixed.

I also changed the call to SetLogger to use the values from the NATS options. This should make the call more clear and behave exactly as ConfigureLogger would.

**Review by S7evinK — APPROVED** [review]
Looks good! Thanks!

**Pull request merged**